### PR TITLE
make sure there are no unknown named arguments fix #140

### LINKED ISSUE #140: .topenv cannot be supplied to log_info()
```r
log_info("...", .topenv = .GlobalEnv)
# Error in log_level(INFO, ..., .logcall = sys.call(), .topcall = sys.call(-1),  : 
#   formal argument ".topenv" matched by multiple actual arguments
```

This looks to be WAI based on the documentation of `...`, but still threw me off. Maybe a friendlier error here? Perhaps if there are any named arguments to `...`?

### PR DESCRIPTION
**Pull request by daroczig**

### CODE REVIEW

File: R/logger.R [hunk 1]
```diff
@@ -311,32 +311,47 @@ validate_log_level <- function(level) {
 }
 
 
+#' Make sure there are no unknown named arguments
+#' @keywords internal
+assert_no_unknown_args <- function() {
+    call <- match.call(definition = sys.function(1), call = sys.call(1), expand.dots = FALSE)
+    stopifnot("Unknown named argument provided." = is.null(names(call$`...`)))
```
Comment (MichaelChirico) [inline R/logger.R:318]: maybe better to supply the aegument(s)?

```suggestion
    if (!is.null(unknown_names <- names(call$`...`))) {
      stop("Unknown argument(s) provided: ", toString(unknown_names))
    }
```

**Comment by daroczig** [comment]
After taking a look at the test results, and thinking this through more, I think failing on named args is not a viable solution, as there are many valid cases to pass named args, e.g. when using JSON loggers, or the below dummy example:

```r
log_info('Hello, {v}!', v = 'world')
INFO [2024-02-29 00:26:30] Hello, world!
```

**Comment by MichaelChirico** [comment]
hmm, in that case, maybe it's the description of '...' that's misleading?

https://github.com/daroczig/logger/blob/ec619ce09d4db7c428745f8152a05907b92cdae9/R/logger.R#L214

**Comment by daroczig** [comment]
I'm not sure if that's misleading, but suggestions welcomed. Formatters take the args and convert to string. Some formatters support named args .. but those are still R objects. If it's confusing, pls suggest better language 🙏

**Comment by MichaelChirico** [comment]
My initial read is that something like `do.call(paste, within(lapply(list(...), as.character), sep = "")` would be done. Indeed that appears the case for un-named entries to `...`. But named arguments are handled differently -- they're used as an environment for substituting values into the unnamed parts of `...` (much like `glue()`). So this distinction of named/unnamed parts of `...` is totally absent.

**Pull request closed without merging**